### PR TITLE
nextdns: init at 1.7.0

### PR DESCRIPTION
**Pull request by pnelson**
###### Motivation for this change

Include the NextDNS DNS/53 to DoH Proxy CLI

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Emantor — APPROVED** [review]
Builds and correctly resolves `google.de` on NixOS unstable.

File: pkgs/applications/networking/nextdns/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "nextdns";
+  version = "1.4.36";
+
+  goPackagePath = "github.com/nextdns/nextdns";
+
```
Comment (jonringer) [inline pkgs/applications/networking/nextdns/default.nix:8]: buildGoModule doesn't use goPackagePath
```suggestion
```

**Review by jonringer — COMMENTED** [review]
Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the commits.

This can be done with:
```
git reset HEAD~1
git add -- pkgs/
git commit --amend --no-edit
git push ... ... --force
```

**Comment by pnelson** [comment]
Updated again with build flags to set the version so it doesn't read "dev".

```sh
$ result/bin/nextdns version
nextdns version 1.4.36
```

**Review by Ma27 — CHANGES_REQUESTED** [review]
You may want to write a simple nixos module which installs the service files for systemd.

**Review by jonringer — COMMENTED** [review]
I would do 2 commits:
```
nextdns: init at 1.5.4
nixos/nextdns: init
```

cc @Infinisil for module review

File: nixos/modules/services/networking/nextdns.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  config = mkIf cfg.enable {
+    systemd.services.nextdns = {
+      description = "NextDNS DNS/53 to DoH Proxy";
+      serviceConfig = {
+        StartLimitInterval = 5;
+        StartLimitBurst = 10;
+        Environment = "SERVICE_RUN_MODE=1";
+        ExecStart = "${pkgs.nextdns}/bin/nextdns run ${config.services.nextdns.arguments}";
+        RestartSec = 120;
+        LimitMEMLOCK = "infinity";
```
Comment (Infinisil) [inline nixos/modules/services/networking/nextdns.nix:33]: Why is this needed? A comment would be nice

File: nixos/modules/services/networking/nextdns.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  options = {
+    services.nextdns = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether or not to enable the NextDNS DNS/53 to DoH Proxy service.";
+      };
+      arguments = mkOption {
+        type = types.separatedString " ";
+        example = "-config 10.0.3.0/24=abcdef";
```
Comment (Infinisil) [inline nixos/modules/services/networking/nextdns.nix:17]: I suggest making this a `types.listOf types.str` instead, then you can use `lib.escapeShellArgs` in the `ExecStart` below not have to worry about shell escaping

File: nixos/modules/services/networking/nextdns.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+let
+  cfg = config.services.nextdns;
+in {
+  options = {
+    services.nextdns = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether or not to enable the NextDNS DNS/53 to DoH Proxy service.";
```
Comment (Infinisil) [inline nixos/modules/services/networking/nextdns.nix:13]:
```suggestion
        description = "Whether to enable the NextDNS DNS/53 to DoH Proxy service.";
```

**Review by infinisil — CHANGES_REQUESTED** [review]

File: nixos/modules/services/networking/nextdns.nix (hunk 2 above)
Comment (pnelson) [reply nixos/modules/services/networking/nextdns.nix:33]: I have no idea. I copied the template from the application source code. I'll add a comment.

https://github.com/nextdns/nextdns/blob/628ea509eaaccd27adb66337db03e5b56f6f38a8/host/service/systemd/service.go

File: nixos/modules/services/networking/nextdns.nix (hunk 3 above)
Comment (pnelson) [reply nixos/modules/services/networking/nextdns.nix]: Hey that's pretty cool - thanks!

File: nixos/modules/services/networking/nextdns.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  };
+
+  # https://github.com/nextdns/nextdns/blob/628ea509eaaccd27adb66337db03e5b56f6f38a8/host/service/systemd/service.go
+  config = mkIf cfg.enable {
+    systemd.services.nextdns = {
+      description = "NextDNS DNS/53 to DoH Proxy";
+      serviceConfig = {
+        StartLimitInterval = 5;
+        StartLimitBurst = 10;
+        Environment = "SERVICE_RUN_MODE=1";
```
Comment (Ma27) [inline nixos/modules/services/networking/nextdns.nix:31]: To override, we probably want to use `systemd.services.nextdns.environment.SERVICE_RUN_MODE = 1` here.

**Review by Ma27 — APPROVED** [review]

**Comment by peterhoeg** [comment]
Thanks for all your hard work on this @pnelson.
